system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.hpp

```diff
@@ -75,21 +78,10 @@ class PeriodicMeasurementNode : public system_metrics_collector::Collector, publ
   virtual void performPeriodicMeasurement();
 
   /**
-   * Creates a ROS2 timer with a period of measurement_period_.
-   *
-   * @return if setup was successful
-   */
-  bool setupStart() override;
-
-  /**
-   * Stops the ROS2 timer
-   *
-   * @return if teardown was successful
+   * Called via a ROS2 timer per the publish_period_. This publishes the statistics derived from
+   * the collected measurements
    */
-  bool setupStop() override;
-
-  // todo implement on publish timer callback, check if we need to clear the window
-  // todo this is part of the publishing interface
+  virtual void publishStatistics() = 0;
```

Comment:
This isn't following the design we discussed or outlined in the doc. Where is the publisher interface? Ideally the PR would be small in the following roll-out:

* introduce the publisher interface and helper methods for message construction
* add implementation in measurement classes

Reply:
Done.